build/release/cloudbuild.yaml

```diff
@@ -26,21 +53,96 @@ steps:
     id: build-make-docker
     args: ["build", "-f", "Dockerfile.build", "-t", "make-docker", "."]
 
+  #
+  # pull the main build image if it exists
+  #
   - name: "make-docker"
     id: pull-build-image
     dir: "build"
     env:
-      - "REGISTRY=us-docker.pkg.dev/${PROJECT_ID}/ci"
+      - "REGISTRY=${_REGISTRY}/ci"
     args: ["pull-build-image"]
     waitFor:
       - build-make-docker
 
+  #
+  # Ensure example images exists
+  #
   - name: "make-docker"
     id: test-examples-on-gar
     dir: "build"
     env:
-      - "REGISTRY=us-docker.pkg.dev/${PROJECT_ID}"
+      - "REGISTRY=${_REGISTRY}"
       - DOCKER_RUN_ARGS=--network=cloudbuild
     args: ["test-examples-on-gar"]
 
-timeout: 3600s
+  #
+  # Build all the images
+  #
+  - name: "make-docker"
+    id: build-images
+    dir: "build"
+    env:
+      - "VERSION=${_VERSION}"
+      - "REGISTRY=${_REGISTRY}/release"
+      - "FULL_BUILD=1"
+    args: ["-j", "1", "build"]
+
+  #
+  # push all the images
+  #
+  - name: "make-docker"
+    id: push-images
+    waitFor:
+      - build-images
+    dir: "build"
+    env:
+      - "VERSION=${_VERSION}"
+      - "REGISTRY=${_REGISTRY}/release"
+    args: ["-j", "1", "push"]
```

Comment:
Could we do this in parrallel? (i.e. `-j=4` (or whatever number is the best)? Maybe make things a bit faster?)